### PR TITLE
[master <  T0984-MG] Upgrade Antrl to 4.10.1 and remove antlr_lock

### PR DESCRIPTION
**Pull request by gitbuda**
- [x] Test + benchmark thoroughly because of the concurrency change
- [x] Upload Antlr generator to the local CI cache

### CODE REVIEW

File: src/query/frontend/ast/cypher_main_visitor.cpp [hunk 1]
```diff
@@ -1967,57 +1979,60 @@ antlrcpp::Any CypherMainVisitor::visitAtom(MemgraphCypher::AtomContext *ctx) {
     // functionInvocation and atom producions in opencypher grammar.
     return static_cast<Expression *>(storage_->Create<Aggregation>(nullptr, nullptr, Aggregation::Op::COUNT));
   } else if (ctx->ALL()) {
-    auto *ident =
-        storage_->Create<Identifier>(ctx->filterExpression()->idInColl()->variable()->accept(this).as<std::string>());
-    Expression *list_expr = ctx->filterExpression()->idInColl()->expression()->accept(this);
+    // TODO(gitbuda): auto or auto* (just to be consistent, I think auto is also fine here).
```
Comment (gitbuda) [inline src/query/frontend/ast/cypher_main_visitor.cpp:1982]: `auto *` seems better because any issue will be detected very early

**Comment by gitbuda** [comment]
`./benchmark.py --num-workers-for-benchmark 16 --export-results benchmark_result.json pokec/medium/*/*` passed on AMD Ryzen 7 3800X 8-Core Processor

**Comment by gitbuda** [comment]
Dedicated parser stress test (#463) with 16 server and 16 client threads also passed 💪

**Review by jbajic — APPROVED** [review]
Just a small comment unrelated to this PR, but to open a new one

File: libs/setup.sh [hunk 2]
```diff
@@ -130,7 +130,7 @@ declare -A primary_urls=(
 # should fail.
 declare -A secondary_urls=(
   ["antlr4-code"]="https://github.com/antlr/antlr4.git"
-  ["antlr4-generator"]="http://www.antlr.org/download/antlr-4.9.2-complete.jar"
+  ["antlr4-generator"]="https://www.antlr.org/download/antlr-4.10.1-complete.jar"
```
[the remark this answers is not in the document]
Comment (gitbuda) [reply libs/setup.sh:133]: The file is in the cache!

**Pull request merged**